sentence_transformers/cross_encoder/CrossEncoder.py

```diff
@@ -185,10 +184,9 @@ def fit(
         train_dataloader.collate_fn = self.smart_batching_collate
 
         if use_amp:
-            if is_torch_npu_available():
-                scaler = torch.npu.amp.GradScaler()
-            else:
-                scaler = torch.cuda.amp.GradScaler()
```

Comment:
I think this if-else is totally fine; it's much simpler than the `device_mod` solution (which also causes a crash if the if-statement is not True).

Could you perhaps revert this change? I think the PR is looking good otherwise.

Reply:
I've reverted this change. @tomaarsen please take a another look, thx :-)